Framework/Core/include/Framework/BinningPolicy.h

```diff
@@ -37,11 +37,43 @@ std::array<std::shared_ptr<arrow::Array>, sizeof...(Cs)> getChunks(arrow::Table*
   return std::array<std::shared_ptr<arrow::Array>, sizeof...(Cs)>{o2::soa::getIndexFromLabel(table, Cs::columnLabel())->chunk(ci)...};
 }
 
-template <typename... Cs>
-std::tuple<typename Cs::type...> getRowData(arrow::Table* table, pack<Cs...>, uint64_t ci, uint64_t ai)
+template <typename C>
+typename C::type getSingleRowPersistentData(arrow::Table* table, uint64_t ci, uint64_t ai)
 {
-  static_assert(std::conjunction_v<typename Cs::persistent...>, "BinningPolicy: only persistent columns accepted (not dynamic and not index ones");
-  return std::make_tuple(std::static_pointer_cast<o2::soa::arrow_array_for_t<typename Cs::type>>(o2::soa::getIndexFromLabel(table, Cs::columnLabel())->chunk(ci))->raw_values()[ai]...);
+  return std::static_pointer_cast<o2::soa::arrow_array_for_t<typename C::type>>(o2::soa::getIndexFromLabel(table, C::columnLabel())->chunk(ci))->raw_values()[ai];
+}
+
+template <typename T, typename C>
+typename C::type getSingleRowDynamicData(T& rowIterator, uint64_t globalIndex)
```

Comment:
This is not universal - there are dynamic columns that require free arguments and attempting to call them will not compile. Probably we can add a dependent type or constexpr int in the dynamic column definition that determines if this particular column needs extra arguments. Then we'll be able to exclude it from this call.

Reply:
Isn't it OK if in the case someone requests such a column, the code doesn't compile?

Reply:
I think it is OK if we do not support those cases for mixing (people can always create a transient table if those columns should really be needed for mixing)

Reply:
This code is useful not only for mixing, getting the content from the whole row can also be used for skimming or for feeding the table to a model. I agree, however, that It would make sense to have a special case for columns with arguments and not just skip them.

Reply:
Well, if this code would be used also outside mixing, then I can move these functions to ASoA.h, for example.
Then, in another PR one could add the support for columns with arguments.

Reply:
I agree

Reply:
OK, so I moved the codes, let me know @aalkin if you'd prefer to see them in another place in ASoA.h